libres/lib/analysis/update.cpp

```diff
@@ -107,12 +107,37 @@ class update_data_type {
     matrix_type *D;
     matrix_type *R;
     matrix_type *A;
-    const bool_vector_type *obs_mask;
+    const std::vector<bool> obs_mask;
     std::vector<std::pair<matrix_type *, std::shared_ptr<RowScaling>>>
         A_with_rowscaling;
     bool has_observations = false;
 };
 
+int_vector_type *
+stl_bool_vector_alloc_active_list(const std::vector<bool> &bool_vector) {
+    int_vector_type *active_list = int_vector_alloc(0, 0);
+    for (int i = 0; i < bool_vector.size(); i++) {
+        if (bool_vector[i])
+            int_vector_append(active_list, i);
+    }
+    return active_list;
+}
+
+int_vector_type *
+stl_bool_vector_alloc_active_index_list(const std::vector<bool> &bool_vector,
```

Comment:
This function is always called with `default_value=-1`- I would just inline that value.